src/classes/GE_GiftEntryController.cls

```diff
@@ -60,63 +64,74 @@ public with sharing class GE_GiftEntryController {
+    }
+
+    /***
+    * @description Sends the purchase request to Elevate and logs an error on a timeout error.
+    * This method is used in tests instead of the sendPurchaseRequest() in order to skip JSON.deserialize(response)
+    * due to it erroring out because the response header contains the Object data type.
+    * @param requestBodyParameters: JSON containing parameters for the purchase call request body
+    * @param dataImportRecordId: Record ID of the related Data Import
+    * @return UTIL_Http.Response
+    */
+    @TestVisible
+    private static UTIL_Http.Response processPurchaseRequest(String requestBodyParameters, String dataImportRecordId) {
+        UTIL_Http.Response response = new UTIL_Http.Response();
+
         try {
             GE_PaymentServices paymentServices = new GE_PaymentServices();
-            paymentServices.setPurchaseCallBody(requestBodyParameters);
-            GE_PaymentServices.ResponseWrapper response = paymentServices.createTransaction();
+            response = paymentServices.createTransaction(requestBodyParameters);
 
-            return JSON.serialize(response);
         } catch (Exception ex) {
-            Boolean isCalloutException = ex instanceof System.CalloutException;
-            Boolean hasTimedOut = isCalloutException && ex.getMessage().contains(TIMED_OUT);
+            String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
+            throw buildDmlException(JSONExceptionData);
 
-            if (hasTimedOut) {
-                return getPurchaseCallTimeoutResponse(ex, dataImportRecordId);
-            } else {
-                String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
-                throw returnDmlException(JSONExceptionData);
-            }
+        } finally {
+            processPurchaseTimeoutResponse(response, dataImportRecordId);
         }
+
+        return response;
     }
 
     /*******************************************************************************************************
-    * @description Creates an http request wrapper for a timeout
-    *
-    * @param ex: A timeout System.CalloutException
+    * @description Updates the timeout response so it can be read by the GE forms
+    * @param response: timeout response
     * @param dataImportRecordId: Record ID of the related Data Import to include in our error logs
-    *
-    * @return String: http response wrapper
+    * @return void
     */
-    private static String getPurchaseCallTimeoutResponse(Exception ex, String dataImportRecordId) {
-        GE_PaymentServices.ResponseWrapper response = new GE_PaymentServices.ResponseWrapper();
-        response.statusCode = TIMEOUT_STATUS_CODE;
-        response.status = ex.getMessage();
+    private static void processPurchaseTimeoutResponse(UTIL_Http.Response response, String dataImportRecordId) {
+        if (response.statusCode != UTIL_Http.STATUS_CODE_REQUEST_TIMEOUT) {
```

Comment:
I think the description block of the header for this method should be updated to include this behavior, that the method returns without further action when the response status code is not STATUS_CODE_REQUEST_TIMEOUT